woocommerce/woocommerce-ios
# Dynamic Dashboard: Share Your Store Card

Part of #12508


## Description


This PR adds the "Share Your Store" card, that is shown when a store has no orders yet. It also adds the logic for:
- checking whether a site has orders or not,
- hiding analytics cards (Performance and Top Performers) if there are no orders yet
- setting the analytics cards above as unavailable in the Dashboard's setting panel
- showing the Share Your Store card if the store has orders and there is a site URL
- doing the actual sharing

Some design decisions:
- While this is also called a card, technically it doesn't belong like the rest of the cards. So, it can't be re-ordered, and doesn't even appear in the Dashboard settings screen.
- It also doesn't have the hide option like the rest of the cards. It disappears automatically once the site has at least one order.

## Testing instructions


1. Log out from app,
2. Login and open a site with no orders yet. I find that using a new Jurassic Ninja site is easiest,
3. Once logged in, ensure the Dashboard shows the "Share Your Store" card and does not show the two analytics cards,
4. Tap the "Share Your Store" button and ensure it opens the share sheet with the site URL,
5. Check settings in Dashboard, ensure the two analytics cards are shown as unavailable
6. Log out from app and login to a site that already has orders, or switch site to that if you have it,
7. Once logged in/switched, ensure the Dashboard does not show the "Share Your Store" card, and shows the analytics cards
8. Check settings in Dashboard, ensure the two analytics cards are not shown as unavailable

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/266376/8e0eb699-43b6-48f1-8390-c00b8a6fe82f" width="42%">
